index.js

```diff
@@ -15,26 +15,30 @@ function JMON(data) {
     type === '[object Array]'
   );
 
-  if (!isObject) return;
+  if (!isObject){
+    this.initialData = data;
+    this.data = data;
+    return;
+  }
 
   this.initialData = data;
   this.data = _.cloneDeep(this.initialData);
 
-  this.isCreated = false;
-  this.isUpdated = false;
-  this.isDeleted = false;
-
   if (isObject){
     var keys = Object.keys(this.data);
-    // intialize all the properties
-    this.set = set;
-    this.get = get;
-    this.commit = commit;
 
+    // intialize all the properties
     keys.forEach((key) => {
       this.data[key] = new JMON(this.data[key]);
```

Comment:
@arshadkazmi42  should we prevent creating new JMOn instance for primitive  types here.

Reply:
Yes. 
I think only JMON instance should be created for objects and not for primitive types.
as none of our functionality will work on primitive types.

Now I get it why are you using the above code of `!obbject`

Reply:
@arshadkazmi42  checked upadted this feature